File: core/ProjectionAreaOf3dShapes/ProjectionAreaOf3dShapes.py
from typing import List


def projectionArea(grid: List[List[int]]) -> int:
    rows, cols = len(grid), len(grid[0])
    row_max, col_max = [0] * rows, [0] * rows
    res = 0
    for row in range(rows):
        for col in range(cols):
            if grid[row][col] != 0:
                res += 1
                row_max[row] = max(row_max[row], grid[row][col])
                col_max[col] = max(col_max[col], grid[row][col])

    return res + sum(row_max) + sum(col_max)

File: core/ProjectionAreaOf3dShapes/ProjectionAreaOf3dShapes.java
package com.dbc;

import java.util.Arrays;

public class ProjectionAreaOf3dShapes {
    public int projectionArea(int[][] grid) {
        int[] rowMax = new int[grid.length];
        int[] colMax = new int[grid.length];
        Arrays.fill(rowMax, 0);
        Arrays.fill(colMax, 0);
        int res = 0;
        for (int i = 0; i < grid.length; i++) {
            for (int j = 0; j < grid[0].length; j++) {
                if (grid[i][j] != 0) {
                    res++;
                    rowMax[i] = Math.max(rowMax[i], grid[i][j]);
                    colMax[j] = Math.max(colMax[j], grid[i][j]);
                }
            }
        }
        for (int val : rowMax) res += val;
        for (int val : colMax) res += val;
        return res;
    }
}
